cve_bin_tool/input_engine.py

```diff
@@ -62,6 +67,51 @@ def input_json(self) -> None:
 
             self.parse_data(set(json_data[0].keys()), json_data)
 
+    def input_vex(self) -> None:
+        analysis_state = {
+            "under_review": Remarks.Unexplored,
+            "in_triage": Remarks.Unexplored,
+            "exploitable": Remarks.Confirmed,
+            "not_affected": Remarks.Mitigated,
+        }
+        with open(self.filename) as json_file:
+            json_data = json.load(json_file)
+            # Only handle CycloneDX VEX file format
+            if json_data["bomFormat"] == "CycloneDX":
+                for vulnerability in json_data["vulnerabilities"]:
+                    id = vulnerability["id"]
+                    state = analysis_state[vulnerability["analysis"]["state"].lower()]
+                    comments = vulnerability["analysis"]["detail"]
+                    for rating in vulnerability["ratings"]:
+                        severity = rating["severity"]
+                    for affect in vulnerability["affects"]:
+                        ref = affect["ref"]
+                        version = None
+                        vendor = None
+                        if "#" in ref:
+                            # Extract product information after # delimiter
+                            p = ref.split("#")[1]
+                            # Last element is version, rest is product which may include -
+                            elements = p.rsplit("-", 1)
+                            product = elements[0]
```

Comment:
Do we want to do a bit of extra input validation here?  I think product may be required to be alphanumeric in NVD but I can't find it in the docs so I could be wrong.  I can't think of any I've seen that have quotes or backslashes, but it's possible that accents and other UTF-8 is in scope?

Reply:
@terriko An interesting comment and observation. I have come across alphanumeric (tends to be lowercase only) and other characters including '-' and '_'. I have also seen '\' used, presumably for escaping purposes. However I suspect validating the characters within a product name would apply in many other places within the tool. I will add something to the input engine and we can then refine it as appropriate.

Reply:
I also need to update the documentation to describe the 'vex' process